lib/components/Input/generic-wheel-picker/index.tsx

```diff
@@ -0,0 +1,216 @@
+import React, {
+    KeyboardEvent,
+    MouseEvent,
+    ComponentProps,
+    forwardRef,
+    useEffect,
+    useRef,
+    useState,
+} from "react";
+import Input, { InputProps } from "../base";
+import { useDropdown } from "@hooks/useDropdown";
+import clsx from "clsx";
+import { DropdownProvider } from "@providers/dropdown/dropdownProvider";
+import useBreakpoints from "@hooks/useBreakpoints";
+import ActionSheet from "@components/ActionSheet";
+import { KEY } from "@utils/common-utils";
+import WheelPickerContainer from "@components/Atom/WheelPicker";
+import "./generic-wheel-picker.scss";
+
+export interface TGenericWheelPickerContent extends InputProps {
+    onClickDropdown?: (e: React.MouseEvent<HTMLDivElement>) => void;
+    containerClassName?: string;
+    actionSheetFooter?: ComponentProps<typeof ActionSheet.Footer>;
+    data: { value: string | number }[][];
+    values: string[] | number[];
+    onValueChange?: (value: (string | number)[]) => void;
+}
+
+const GenericWheelPickerContent = forwardRef<
+    HTMLDivElement,
+    TGenericWheelPickerContent
+>(
+    (
+        {
+            children,
+            className,
+            values = [],
+            onClickDropdown,
+            onValueChange,
+            containerClassName,
+            actionSheetFooter,
+            label,
+            leftIcon,
+            rightIcon,
+            ...rest
+        },
+        ref,
+    ) => {
+        const containerRef = useRef<HTMLDivElement>(null);
+        const actionSheetRef = useRef<HTMLDivElement>(null);
+
+        const [inputValues, setInputValues] =
+            useState<(string | number)[]>(values);
+        const [isPressed, setIsPressed] = useState(false);
+
+        const handleMouseDown = () => {
+            setIsPressed(true);
+        };
+
+        const handleMouseUp = () => {
+            setIsPressed(false);
+        };
+
+        const handleMouseLeave = () => {
```

Comment:
handleMouseUp  and handleMouseLeave do the same and all of 3 functions can be passed to the props as () => setIsPressed(...)